### ISSUE TITLE
Legion: independent dimensions of datasets with respect to index/field space dimensions

### ISSUE THREAD

**Issue by bandokihiro** [StanfordLegion/legion#747]
In the context of a CFD simulation, my index space is a 1D `Rect` of size number of elements. In order to write solution files for specified physical variables, I need to write multiple doubles per variable per element.

For now, let's say my discretization requires me to write 4 doubles per variable per element. In the current Legion's implementation, I need to create the following HDF5 datatype
```cpp
ArrayType dtype(PredType::NATIVE_DOUBLE, 1, 4);
```

If I was able to tell my visualization tool (paraview) how to read it, I would not have any problem. However, currently, I have not found any other way than pre-processing the files to unpack every array type to single doubles.

Desired features:
- ability to create datasets of size not necessarilly equal to the dimension of the index space (in the example, create datasets of size `4*nElem` rather than `nElem` and packing doubles by 4)
- ability to create datasets of size not necessarilly related to the dimensions of the index/field space (for instance, I would like to create dataset of size `{4*neElem, dim}` to store the mesh coordinates)

**Comment by lightsighter** [comment]
I'm not sure I fully understand your problem. Fields in Legion can have an arbitrary size and don't need to be a plain-old-datatype. You can store things like `double[4]` in a field without any trouble (you might also do it with a `std::array<double,4>`). This should work correctly with HDF5 instances as well.

I think having a concrete code example here to look at would help us resolve the ambiguity.

**Comment by bandokihiro** [comment]
Ok I will try to do that. Meanwhile the problem comes from the fact that the index space and the HDF5 dataset need to be of the same size (I think, according to a discussion I had with Sean). So if my index space is of type `Rect` of size 10 and the field space is of type `double[4]`, then I need to create an HDF5 dataset of size 10 with the `ArrayType` I mentioned previously instead of having a dataset of size 40 of type `double`.

**Comment by lightsighter** [comment]
Yes, I believe that constraint is still accurate. There's no way to "reinterpret" the structure of your data from HDF5 to Legion/Realm at the moment I believe. @streichler to comment if that is true in Realm. If it's not true in Realm, I should be able to finagle things at the Legion level to get something to work, but if Realm doesn't support it then there's not much I can do without some changes in Realm.
